### PR TITLE
JP-3593: 2nd group saturation part 2

### PR DESCRIPTION
**Pull request by drlaw1558**
Ready for review: this addresses JP-3593, flagging cosmic ray saturation in group 2 data in a consistent manner between both IRS2 and non-IRS2 modes using the new use_readpatt keyword.  This PR is tied to STCAL PR https://github.com/spacetelescope/stcal/pull/283

**Checklist for PR authors (skip items if you don't have permissions or they are not applicable)**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [x] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [x] All comments are resolved
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by drlaw1558** [comment]
@tapastro Should now be ready for review.

File: jwst/saturation/saturation.py [hunk 1]
```diff
@@ -84,7 +93,7 @@ def flag_saturation(input_model, ref_model, n_pix_grow_sat):
     return output_model
 
 
-def irs2_flag_saturation(input_model, ref_model, n_pix_grow_sat):
+def irs2_flag_saturation(input_model, ref_model, n_pix_grow_sat, use_readpatt):
```
Comment (melanieclarke) [inline jwst/saturation/saturation.py:96]: Add use_readpatt to parameters in docstring

File: jwst/saturation/saturation.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
 ATOD_LIMIT = 65535.  # Hard DN limit of 16-bit A-to-D converter
 
 
-def flag_saturation(input_model, ref_model, n_pix_grow_sat):
+def flag_saturation(input_model, ref_model, n_pix_grow_sat, use_readpatt):
```
Comment (melanieclarke) [inline jwst/saturation/saturation.py:25]: Add use_readpatt to parameters in docstring

**Review by melanieclarke — COMMENTED** [review]
This looks good to me, given that Roman is happy with the corresponding changes in stcal, and given your testing across multiple instrument modes.  Thanks for the unit tests and documentation updates!

One small documentation request below, for the new function parameters.

I will run regression tests when the system is free.

File: jwst/saturation/saturation.py (hunk 1 above)
Comment (drlaw1558) [reply jwst/saturation/saturation.py:99]: Done!

File: jwst/saturation/saturation.py (hunk 2 above)
Comment (drlaw1558) [reply jwst/saturation/saturation.py:25]: Done!

File: jwst/saturation/saturation.py [hunk 3]
```diff
@@ -41,6 +41,9 @@ def flag_saturation(input_model, ref_model, n_pix_grow_sat):
         as saturated (i.e '1' will flag the surrouding 8 pixels) to account for
         charge spilling.
 
+    use_readpatt: boolean
```
Comment (melanieclarke) [inline jwst/saturation/saturation.py:44]:
```suggestion
    use_readpatt : bool
```

File: jwst/saturation/saturation.py [hunk 4]
```diff
@@ -106,6 +118,9 @@ def irs2_flag_saturation(input_model, ref_model, n_pix_grow_sat):
...
```
Comment (melanieclarke) [inline jwst/saturation/saturation.py:121]:
```suggestion
    use_readpatt : bool
```

**Review by melanieclarke — COMMENTED** [review]
Thanks for the update!  One last small formatting thing - the type annotation should be the Python type (bool).

I started regression tests here:
https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1709/

**Review by melanieclarke — APPROVED** [review]
In the regtests, I see a difference for NIRISS rate files, which I think is because the corresponding stcal branch is out of date - the difference looks related to the chargeloss readnoise variance calculation change.

I also see one difference for a NIRSpec rate file, to the DQ plane only.  It looks like some pixels that were do not use but not flagged as saturated are now also flagged as saturated.  Looking at the image in a local run, it certainly looks like they were saturated, so this is a reasonable change.

I think this is good to go from my perspective, pending the merge of the corresponding stcal PR.  @tapastro - can you please coordinate on the stcal side?

**Pull request merged**